test/test_components.py

```diff
@@ -1449,6 +1449,27 @@ def test_postprocess(self):
         result_ = component.postprocess({"text": text, "entities": entities})
         assert result == result_
 
+        # Test split entity is merged when combine adjacent is set
+        text = "Wolfgang lives in Berlin"
+        entities = [
+            {"entity": "PER", "start": 0, "end": 4},
+            {"entity": "PER", "start": 4, "end": 8},
+            {"entity": "LOC", "start": 18, "end": 24},
+        ]
+        # After a merge empty entries are stripped except the leading one
+        result_after_merge = [("", None)] + [
+            entry for entry in result if entry != ("", None)
+        ]
+        default_adjacent = component.combine_adjacent
+        component.combine_adjacent = False
+        result_ = component.postprocess({"text": text, "entities": entities})
+        assert result != result_
+        assert result_after_merge != result_
+        component.combine_adjacent = True
+        result_ = component.postprocess({"text": text, "entities": entities})
+        assert result_after_merge == result_
+        component.combine_adjacent = default_adjacent
```

Comment:
Instead of changing the attribute of the component, simply create a new component when necessary
```suggestion
        result_ = component.postprocess({"text": text, "entities": entities})
        assert result != result_
        assert result_after_merge != result_
        
        component = gr.HighlightedText(combine_adjacent=True)
        result_ = component.postprocess({"text": text, "entities": entities})
        assert result_after_merge == result_
        
        component = gr.HighlightedText(combine_adjacent=True)
```